HIP/hip-1056.md

```diff
@@ -0,0 +1,1727 @@
+
+    /**
+     * All Token transfers as a result of this transaction.
+     */
+    repeated proto.TokenTransferList token_transfer_lists = 8;
+
+    /**
+     * All token associations implicitly created while handling this transaction.
+     */
+    repeated proto.TokenAssociation automatic_token_associations = 9;
+
+    /**
+     * List of accounts with the corresponding staking rewards paid as a result of
+     * a transaction.
+     */
+    repeated proto.AccountAmount paid_staking_rewards = 10;
+
+    /**
+     * Congestion pricing multiplier at the time the transaction was executed.
+     */
+    uint64 congestion_pricing_multiplier = 11;
+
+    /**
+     * A [string, code] to indicate what, specifically, produced
+     * a non-success response code.
+     */
+    //  bytes error_message = 12;
+    // Alternative
+    //  uint16 detail_error_code = 12;
+}
+```
+
+<aside>
+🚨 **Open Decision**: Decide on string error message or a integer detailed error code?
+
+</aside>
+
+<aside>
+🚨 **Open Question**: We discussed if TransferList should be included or not here. I don’t remember exactly why it has
+to be. We should double check if it has to be and can not be removed and if it has to be here then document why.
+ - The Mirror Node MAPI exposes the transfer list, and it’s used by clients of mirror node.
+ - There are some transfers only exposed in the transfer list currently.
+</aside>
+
+#### Transaction Output
+
+Output data from a transaction data that is neither present in the transaction nor stored in state. Not all transactions
+have output, each one that does has a corresponding output message type with any data it needs to include.
+
+<aside>
+🚨 **Open Task: Update section with missing TransactionOutput objects**.
+</aside>
+
+```protobuf
+message TransactionOutput {
+    oneof transaction {
+        /**
+         * Output from a crypto transfer transaction.
+         */
+        CryptoTransferOutput crypto_transfer = 1;
+
+        /**
+         * Output from a utilPrng transaction to request a
+         * deterministic pseudo-random number.
+         */
+        UtilPrngOutput util_prng = 2;
+
+        /**
+         * Output from a contract call transaction.
+         */
+        CallContractOutput contract_call = 3;
+
+        /**
+         * Output from an ethereum call transaction.
+         */
+        EthereumOutput ethereum_call = 4;
+
+        /**
+         * Output from a contract create transaction.
+         */
+        CreateContractOutput contract_create = 5;
+
+        /**
+         * Output from a schedule create transaction that executed
+         * immediately on creation.
+         */
+        CreateScheduleOutput create_schedule = 6;
+
+        /**
+         * Output from a schedule sign transaction that resulted in
+         * executing the scheduled transaction.
+         */
+        SignScheduleOutput sign_schedule = 7;
+    }
+}
+```
+
+```protobuf
+message CryptoTransferOutput {
```

Comment:
The crypto create might result in auto account creation. Shouldn't we also add a record for it for those cases?

Reply:
The account creation will be present in StateChanges, no additional output is needed.

Reply:
Need to call out  parent child considerations more explicitly. Might be done but noting here as a reminder to come back in case there's room for improvement